lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1653,6 +1669,255 @@ def mock_certificate_downloadable_status(
+        )
+        if due_date is not None:
+            metadata['due'] = due_date
+
+        if graded:
+            metadata['graded'] = True
+            metadata['format'] = 'Homework'
+
+        with self.store.bulk_operations(self.course.id):
+            self.chapter = ItemFactory.create(category='chapter', parent_location=self.course.location,
+                                              display_name="Section 1")
+            self.section = ItemFactory.create(category='sequential', parent_location=self.chapter.location,
+                                              display_name="Subsection 1", metadata=metadata)
+            self.vertical = ItemFactory.create(category='vertical', parent_location=self.section.location)
+            if add_problem:
+                problem_xml = MultipleChoiceResponseXMLFactory().build_xml(
+                    question_text='The correct answer is Choice 1',
+                    choices=[True, False],
+                    choice_names=['choice_0', 'choice_1']
+                )
+                self.problem = ItemFactory.create(category='problem', parent_location=self.vertical.location,
+                                                  data=problem_xml, display_name='Problem 1')
+
+        CourseEnrollmentFactory(user=self.user, course_id=self.course.id, mode=CourseMode.HONOR)
+
+    def answer_problem(self, value=1, max_value=1):
+        """
+        Make the user answer the problems.
+        """
+        request = get_mock_request(self.user)
+        grade_dict = {'value': value, 'max_value': max_value, 'user_id': self.user.id}
+        field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.user,
+            self.course,
+            depth=2
+        )
+        # pylint: disable=protected-access
+        module = get_module(
+            self.user,
+            request,
+            self.problem.scope_ids.usage_id,
+            field_data_cache,
+        )._xmodule
+        module.system.publish(self.problem, 'grade', grade_dict)
+
+    def assert_progress_page_show_grades(self, response, show_correctness, due_date, graded,
+                                         show_grades, score, max_score, avg):
+        """
+        Ensures that grades and scores are shown or not shown on the progress page as required.
+        """
+
+        expected_score = "<dd>{score}/{max_score}</dd>".format(score=score, max_score=max_score)
+        percent = score / float(max_score)
+        graded_data = [dict(color='#b72121', label='Homework',
+                            data=[[1, percent], [2, 0.0], [3.25, float(avg)]])]
+        if percent > 0:
+            graded_data.append(dict(color='#b72121', data=[[4.75, float(avg)]],
+                                    label="Homework-grade_breakdown"))
+
+        ungraded_data = [dict(color='#b72121', label='Homework',
+                              data=[[1, 0.0], [2, 0.0], [3.25, 0.0]])]
```

Comment:
Did the same for `"Homework"`, addressed with https://github.com/edx/edx-platform/pull/15018/commits/d7e44c50acc6dc3a73fc400f410cfe4ceaca834d#diff-8aefe502dfa8ee5a3e18c9227e61b034R1777.